Extension/src/Debugger/configurationProvider.ts

```diff
@@ -57,9 +56,9 @@ export class QuickPickConfigurationProvider implements vscode.DebugConfiguration
 
         return vscode.window.showQuickPick(items, {placeHolder: "Select a configuration"}).then(async selection => {
             // Wrap in new Promise to make sure task kicks off before VS Code switches the active document to launch.json
-            return new Promise<vscode.DebugConfiguration[]>(async resolve => {
+            return new Promise<vscode.DebugConfiguration[]>(async (resolve, reject) => {
                 if (!selection) {
-                    return resolve([defaultConfig]); // User canceled it. Choose the default config?
+                    return reject([]); // User canceled it.
```

Comment:
If you want to return an empty array, the way to do it is with `resolve([])`, not `reject([])`.  reject goes through the error callback (or throws an exception if you're using `await`)

Reply:
Griffin was saying that was intentional. I'll discuss it with you tomorrow.

Reply:
There is an async anti-pattern in here.  We talked about it and replaced the promises with async/await